File: todo-list-backend/src/main/java/com/todolist/model/Task.java
package com.todolist.model;

import java.time.ZonedDateTime;

public class Task {
	private String id;
	private String title;
	private ZonedDateTime createdAt;
	private TaskStatus status;

	public String getId() {
		return this.id;
	}

	public void setId(final String id) {
		this.id = id;
	}

	public String getTitle() {
		return this.title;
	}

	public void setTitle(final String title) {
		this.title = title;
	}

	public ZonedDateTime getCreatedAt() {
		return this.createdAt;
	}

	public void setCreatedAt(final ZonedDateTime createdAt) {
		this.createdAt = createdAt;
	}

	public TaskStatus getStatus() {
		return this.status;
	}

	public void setStatus(final TaskStatus status) {
		this.status = status;
	}
}

File: todo-list-frontend/src/models/Task.ts

export enum TaskStatus {
  NOT_DONE = 'NOT_DONE',
  DONE = 'DONE',
}

export type TaskId = string;

interface Task {
  id: TaskId;
  title: string;
  status: TaskStatus;
  createdAt: Date;
}

export default Task;
